### PR TITLE
feat: add snapshot timestamp in file name

### PR DESCRIPTION
**Pull request by lichuang**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

1. feat: add snapshot timestamp in [snapshot|segment|block|index] file name
2. move snapshot v5 to v6 since it was not used.

- Closes https://github.com/datafuselabs/databend/issues/12671

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/12754)

### CODE REVIEW

**Comment by JackTan25** [comment]
askbend:summary

**Comment by sundy-li** [comment]
Why not change the style of `snapshot_id` to include the timestamp ?

File: src/query/storages/common/table-meta/src/meta/v5/snapshot.rs [hunk 1]
```diff
@@ -103,6 +102,16 @@ impl TableSnapshot {
         // trim timestamp to micro seconds
         let trimmed_timestamp = trim_timestamp_to_micro_second(adjusted_timestamp);
         let timestamp = Some(trimmed_timestamp);
+        let prev_snapshot_id = match prev_id {
+            Some((prev_snapshot_id, prev_format)) => {
+                if prev_format <= 4 {
+                    Some((None, prev_snapshot_id, prev_format))
```
Comment (JackTan25) [inline src/query/storages/common/table-meta/src/meta/v5/snapshot.rs:108]: If we read an old snapshot, we think the timestamp is very old, and we will vacuum it? How to process the default timestamp, or read it from snapshot,right?

**Comment by zhyass** [comment]
> Why not change the style of `snapshot_id` to include the timestamp ?

I agree with @sundy-li ’s suggestion. Perhaps we can consider using ulid to save space.

**Comment by sundy-li** [comment]
https://github.com/oklog/ulid#string-representation

**Pull request closed without merging**